src/libcore/ptr.rs

```diff
@@ -532,6 +532,15 @@ pub struct Unique<T: ?Sized> {
     _marker: PhantomData<T>,
 }
 
+#[unstable(feature = "unique", issue = "27730")]
+impl<T: fmt::Debug> fmt::Debug for Unique<T> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.debug_struct("Unique")
+            .field("pointer", &self.pointer)
+            .finish()
```

Comment:
Looking at `Box`, it prints the actual value in `Display` and `Debug`, instead of the pointer. I'm thinking just copy that approach, and also implement `fmt::Pointer` as well.